Improve sample config

nirs opened the pull request:
- Don't use commented values since it is hard to uncomment yaml objects                                               
- More clear default values
- Add new name property for clusters
- Add titles for configuration sections using ##
- Document which value may need to be modified or added
- Add missing options: repo, dr-policy, clusterset, tests
- Rename creator to commandName
- Use Go template to render the sample config, preparing for creating sample for drenv environment.

## Example sample config

```yaml
## ramenctl configuration file

## Clusters configuration.
# - Modify clusters "kubeconfig" and "name" to match your hub and managed
#   clusters names and path to the kubeconfig file.
clusters:
  hub:
    name: hub
    kubeconfig: hub/config
  c1:
    name: primary
    kubeconfig: primary/config
  c2:
    name: secondary
    kubeconfig: secondary/config

## Git repository for test command.
# - Modify "url" to use your own Git repository.
# - Modify "branch" to test a different branch.
repo:
  url: https://github.com/RamenDR/ocm-ramen-samples.git
  branch: main

## DRPolicy for test command".
# - Modify to match actual DRPolicy in the hub cluster.
drPolicy: dr-policy

## ClusterSet for test command".
# - Modify to match your Open Cluster Management configuration.
clusterSet: default

## PVC specifications for test command".
# - Modify items "storageclassname" to match the actual storage classes in the
#   managed clusters.
# - Add new items for testing more storage types.
PVCSpecs:
- name: rbd
  storageClassName: rook-ceph-block
  accessModes: ReadWriteOnce
- name: cephfs
  storageClassName: rook-cephfs-fs1
  accessModes: ReadWriteMany

## Tests cases for test command".
# - Modify the test for your preferred workload or deployment type.
# - Add new tests for testing more combinations in parallel.
# - Available workloads: deploy
# - Available deployers: appset, subscr, disapp
tests:
- workload: deploy
  deployer: appset
  pvcSpec: rbd
```

Fixes #21

parikshithb commented on pkg/config/config.go:
```diff
@@ -10,28 +10,38 @@ import (
+# Clusters configurations - modify to match your clusters.
+clusters:
+  hub:
+    kubeconfigpath: hub.kubeconfig			# Add hub cluster kubeconfig
```
spacing here is uneven

nirs replied:
I did not find the issue, I hope it is fixed now.

parikshithb commented on pkg/config/config.go:
```diff
@@ -10,28 +10,38 @@ import (
+# Clusters configurations - modify to match your clusters.
+clusters:
+  hub:
+    kubeconfigpath: hub.kubeconfig			# Add hub cluster kubeconfig
```
maybe from hub.kubeconfig -> hub/kubeconfig? similarly for primary and secondary.

nirs replied:
Changed to "hub/config", matching the default "~/.kube/config". We already mention "kubeconfig" in the name of the option so there is no need to repeat it.

parikshithb commented on pkg/config/config.go:
```diff
@@ -10,28 +10,38 @@ import (
 pvcspecs:
 - name: rbd
-  storageclassname: rook-ceph-block
+  storageclassname: rook-ceph-block			# Add rbd storage class name
```
all inline comments(in pvpspec and clusters) can start from same position to make it look more consistent.

nirs replied:
Eliminated the comment per line.

parikshithb commented on pkg/config/config.go:
```diff
@@ -10,28 +10,38 @@ import (
+  hub:
+    kubeconfigpath: hub.kubeconfig			# Add hub cluster kubeconfig
+  c1:
+    kubeconfigpath: primary.kubeconfig		# Add primary managed cluster kubeconfig
```
Instead of `# Add primary managed cluster kubeconfig`, think it might be better to comment this way 
`# Modify: Path to the primary managed cluster kubeconfig` to make instructions clear. wdyt?

nirs replied:
Modify is better bu the text should be shorter

nirs replied:
Should be addressed in new comments.

parikshithb commented on pkg/config/config.go:
```diff
@@ -4,39 +4,70 @@
+## ClusterSet.
+# - Modify to match your Open Cluster Management configuration.
+clusterSet: default
+
```
Git repo configuration is missing, we need to add it:
In ramen e2e config.yaml.sample, planning to add:
```
repo:
  url: "https://github.com/RamenDR/ocm-ramen-samples.git"
  branch: main
```

nirs commented:
Changes in latest version:
- Split to cleaner commits
- Add missing repo
- Added "for test command" for all sections related only to "ramenctl test"
- Rename config type to Sample
- Nice way to render the sample with Bytes().

parikshithb commented on pkg/config/config.go:
```diff
@@ -4,39 +4,74 @@
+repo:
+  url: https://github.com/RamenDR/ocm-ramen-samples.git
+  branch: main
+
```
We missed namespace configuration. We need to add similar config which we are planning to add in e2e [config](https://github.com/RamenDR/ramen/pull/1883/files#diff-1243b5d5fe88be6e909a5a5ca3be61392484adbd11a6e228e15e8e9345767e8aR10).

nirs replied:
yes but this should really be determined internally, this is too much to ask from a user to change so many things. Making this configurable can be future feature.
